new-frontiers-14/frontier-station-14
# Allow for selective tool disabling

## About the PR



Extends the OnToolsUseComponent to allow for individual tool qualities (e.g. Anchoring, Prying, Pulsing) to be blocked.

Extends the ToolUseAttemptEvent to contain the entity of the tool being used and the qualities of the tool being used in the attempt.

## Why / Balance


Greater flexibility in handling tool actions, more convenient for limiting particularly sensitive actions in a safe zone (e.g. door hacking)

## How to test


1. Spawn a CuddlyCritterVend (should only block anchoring).
2. Try to unanchor it, it should fail almost immediately (before, it would have a doafter that wouldn't finish)
3. Screwdriver it, the panel should open.
4. Cut a wire, it should cut.
5. Spawn a POI IFF console (should block all tool use).
6. Try to unanchor it, it shouldn't work.
7. Try to screwdriver it, it shouldn't work.

## Media



## Breaking changes


**Changelog**



Internal feature, no real gameplay impact.
